### ISSUE TITLE
Replace all string constants with translation keys

### ISSUE THREAD

**Issue by lucaslcode** [Couchers-org/couchers#4152]
We use i18n for translations. In some places in the app, we forgot to use it and have hardcoded strings. This means they won't be translated. This PR is about finding those hardcoded strings and switching them to `t` translation functions or `Trans` components, and adding the translation (start with English for now) to the proper resource json file.

-----------

Overview on how translations work and how to use them for different pages and components:

# 1. Translation keys

We need to replace all string constants with translation keys.

For example, 
```
<HtmlMeta title={CONNECTIONS} />
```
should become
```
import { CONNECTIONS } from "i18n/namespaces";
...
const { t } = useTranslation(CONNECTIONS);
...
<HtmlMeta title={t("connections_page_title")} />
```

These translation keys with strings live in `features/<featureName>/locales/en.json`. 

The translation keys should, as best as possible, describe the location or context of the string, not the text content. An example of a badly named key is `"no_friends": "No friends available",`. A well named key is `"friends_list_empty": "No friends available",`. If we wanted to change the text to "There aren't any friends here yet", `no_friends` would no longer make sense, but `friends_list_empty` does.

The [`features/auth`](https://github.com/Couchers-org/couchers/blob/develop/app/web/features/auth/locales/en.json) folder is mostly complete, with good keys. Use this as a reference on how to implement translation keys.

`react-i18next` can handle [interpolation](https://react.i18next.com/latest/trans-component), [plurals](https://www.i18next.com/translation-function/plurals) and [formatting (https://www.i18next.com/translation-function/formatting) (click links for documentation). [See this file for an example](https://github.com/Couchers-org/couchers/blob/develop/app/web/features/auth/email/ChangeEmail.tsx).

# 2. Use `i18n` abstracted import

We should also **change all imports** of `react-i18next` (or in our app, `next-i18next`) to use our abstracted `i18n` functions introduced in Couchers-org/couchers#2329.  Simply change something like `import { useTranslation } from "next-i18next";` to `import { useTranslation } from "i18n";`.

# 3. Use helper function for `getStaticProps` for `pages` folder

Finally, the actual pages in the `pages` folder include their translations by exporting `getStaticProps` with the translation data. Couchers-org/couchers#2329 also introduced a helper function to make this much cleaner, turning this:
```
export const getStaticProps: GetStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(
      locale ?? "en",
      ["global", "profile"],
      nextI18nextConfig
    )),
  },
});
```
into this: 
```
export const getStaticProps = translationStaticProps([GLOBAL, PROFILE]);
```
So all `getStaticProps` which only include translations (maybe that's all of them?) should use this with the relevant namespaces.

**Comment by lucaslcode** [comment]
Before starting, Couchers-org/couchers#2329 should be merged, and I will also do globals (as every feature may use globals).

**Comment by papeldeorigami** [comment]
I will continue working on the **communities** screens. Since it is a big feature, I started with communities/events - see [Issue 2226](https://github.com/Couchers-org/couchers/pull/2226)
I still have to adjust to this new import pattern.

**hint** for this translation task it is important to run `yarn test-ci` after changes because some eslint errors are not checked with `yarn test`. For instance, references to deleted constants in other feature folders.

**Comment by darrenvong** [comment]
Yeah I think I will check auth and donations since I've done those. After that, I can pick up messages as well.

**Comment by alesdvorakcz** [comment]
I will start with Connections and then anything else what will be unassigned

**Comment by tsuruby** [comment]
I will take Profile.

**Comment by lucaslcode** [comment]
Just note, as you can see in Couchers-org/couchers#2394 there are loads of keys in the global namespace which shouldn't be there, particularly for profiles and communities

**Comment by darrenvong** [comment]
One more thing worth mentioning is that in some of the older UI code, there are still some stragglers which aren't extracted to constants! So if any of you see a hardcoded string in the area you are working in, they should also be extracted into the JSON following the guidelines in this issue.

**Comment by papeldeorigami** [comment]
FYI I will abort the branch I was working with. After trying to rebase it, I decided it is better to start from scratch. I will create one focused **communities/events** only and try to merge it as soon as possible. If the merge is successful, I will proceed with other communities' features.

**Comment by papeldeorigami** [comment]
Communities feature is now ready for merge, so I will proceed with **Donations**. I think I got to speed with these refactoring tasks, so if someone wants me to prioritize some specific domain, please let me know.

**Comment by nabramow** [comment]
@aapeliv do you know if this is still relevant?

**Comment by darrenvong** [comment]
👋 Coming from someone who has a bit of context about this, I'd say this still needs to be done, since I think there are still a small number of hardcoded strings/constants lying around in our codebase that stops the app from being translatable into multiple languages.

We got through about 80% of the app as a rough guess before we lost a bit of steam 😅

**Comment by nabramow** [comment]
Thanks for jumping in @darrenvong . I'll update the description and move it to "Ready" then!

**Comment by nabramow** [comment]
**UPDATE Nov 2025**: There are still a small number of hardcoded strings/constants lying around in our codebase that stops the app from being translatable into multiple languages. We got through about 80% of the app as a rough guess before we lost a bit of steam

So this ticket is mostly checking for what we missed. Leaving the rest here for context.

> # 1. Translation keys
> We need to replace all string constants with translation keys.
> 
> For example,
> 
> ```
> <HtmlMeta title={CONNECTIONS} />
> ```
> 
> should become
> 
> ```
> import { CONNECTIONS } from "i18n/namespaces";
> ...
> const { t } = useTranslation(CONNECTIONS);
> ...
> <HtmlMeta title={t("connections_page_title")} />
> ```
> 
> These translation keys with strings live in `features/<featureName>/locales/en.json`. However:
> 
> * they are mostly not used in the actual code yet
> * they often follow bad naming practices (see below)
> 
> The translation keys should, as best as possible, describe the location or context of the string, not the text content. An example of a badly named key is `"no_friends": "No friends available",`. A well named key is `"friends_list_empty": "No friends available",`. If we wanted to change the text to "There aren't any friends here yet", `no_friends` would no longer make sense, but `friends_list_empty` does.
> 
> The [`features/auth`](https://github.com/Couchers-org/couchers/blob/develop/app/web/features/auth/locales/en.json) folder is mostly complete, with good keys. Use this as a reference on how to implement translation keys.
> 
> `react-i18next` can handle [interpolation](https://react.i18next.com/latest/trans-component), [plurals](https://www.i18next.com/translation-function/plurals) and [formatting (https://www.i18next.com/translation-function/formatting) (click links for documentation). [See this file for an example](https://github.com/Couchers-org/couchers/blob/develop/app/web/features/auth/email/ChangeEmail.tsx).
> 
> # 2. Use `i18n` abstracted import
> We should also **change all imports** of `react-i18next` (or in our app, `next-i18next`) to use our abstracted `i18n` functions introduced in #2329. Simply change something like `import { useTranslation } from "next-i18next";` to `import { useTranslation } from "i18n";`.
> 
> # 3. Use helper function for `getStaticProps`
> Finally, the actual pages in the `pages` folder include their translations by exporting `getStaticProps` with the translation data. #2329 also introduced a helper function to make this much cleaner, turning this:
> 
> ```
> export const getStaticProps: GetStaticProps = async ({ locale }) => ({
>   props: {
>     ...(await serverSideTranslations(
>       locale ?? "en",
>       ["global", "profile"],
>       nextI18nextConfig
>     )),
>   },
> });
> ```
> 
> into this:
> 
> ```
> export const getStaticProps = translationStaticProps([GLOBAL, PROFILE]);
> ```
> 
> So all `getStaticProps` which only include translations (maybe that's all of them?) should use this with the relevant namespaces.
> 
> # Separation of work
> Comment below with what feature you would like to work on:
> 
> * [x]  Auth (this is done but needs checking, maybe @darrenvong do this)
> * [x]  Communities (big)
> * [x]  Connections (small)
> * [x]  Dashboard
> * [x]  Donations (just needs checking)
> * [x]  Messages
> * [x]  Profile
> * [x]  Search
> * [x]  Globals
> * [ ]  Components (part of globals)

**Comment by tristanlabelle** [comment]
@nabramow This feature seems not very actionable. I'd suggest closing it in favor of individual issues when we find nonlocalized strings. I'd also like to implement support for [pseudolocalization](https://en.wikipedia.org/wiki/Pseudolocalization), which would help us to locate any non-localized strings. (filed #7648 )

**Comment by nabramow** [comment]
@tristanlabelle feel free to close this one using dropdown "close as not planned"
